src/components/EndChatButton.tsx

```diff
@@ -10,7 +11,13 @@ type Props = {
 export default function EndChat({ channelSid, token }: Props) {
   const handleEndChat = async () => {
     try {
-      await endChat(channelSid, token);
+      const response: any = await endChat(channelSid, token);
+
+      const { isTaskStageAssigned } = response;
+
+      if (!isTaskStageAssigned) {
+        FlexWebChat.Actions.invokeAction('RestartEngagement');
+      }
```

Comment:
Why is this necessary?

Reply:
Without this change, the state of the channel remains active for the user in webchat in all the task assignment stages. There is indications for the worker in flex. But there is no way to get to 'Start New Chat' for the user without triggering the post survey. 
(Perhaps I should change the channel to `INACTIVE` from serverless).

 Either of these are half measures. Ideally flex (in the future taskroutercallback) will take care of this.